src/vcpkg/commands.integrate.cpp

```diff
@@ -272,51 +272,41 @@ namespace vcpkg::Commands::Integrate
                 switch (user_choice)
                 {
                     case ElevationPromptChoice::YES: break;
-                    case ElevationPromptChoice::NO:
-                        msg::println_warning(msgPreviousIntegrationFileRemains);
-                        Checks::exit_fail(VCPKG_LINE_INFO);
+                    case ElevationPromptChoice::NO: msg::println_warning(msgPreviousIntegrationFileRemains); break;
```

Comment:
Sorry, I didn't explain myself correctly. I did want to propose a change in behavior.

`integrate install` _used to_ perform these steps, in this order:

1- Write a props file to `%PROGRAMFILES(x86)%`.
2- Write props/targets files to `%LOCALAPPDATA%`.
3- Display success and usage messages.

PR #1034 reversed to order of the first two operations. The new order is:

1- Write props/targets files to `%LOCALAPPDATA%`.
2- Write a props file to` %PROGRAMFILES(x86)%`.
3- Display success and usage messages.

This means that a user with no write access to `%PROGRAMFILES(x86)%` now gets a mostly working integration (except for VS2015/MSBuild14).

However, for this mostly happy unprivileged user, the command terminates with an error and doesn't display any usage message or indication of (partial) success.

Perhaps it would be better not to treat a failure to write to `%PROGRAMFILES(x86)%` as a hard error, but instead to report it as a warning, and to print the usage message at the end? This is what the PR is proposing.

There is no observable change if the MSBuild14 directory is writable, or if its content is up to date. Other errors, such as failure to write `%LOCALAPPDATA%`, still result in termination. This is just about being tolerant to FS errors when writing to `%PROGRAMFILES(x86)%`.

Anyway, that was the initial aim. What do you think of this approach?

Personally, after thinking about it for a few days, I don't think that this PR can be applied in its actual state. In fact, I am considering closing it. For instance, saying `All MSBuild C++ projects can now #include any installed libraries...` in the case of partially applied integration is misleading. Unless the messages are reworded, I think that the status quo is better.

Reply:
I think it's a good change to "do what we can", but I just think we should still report a nonzero exit code if there was partial failure.

Agreed that rewording the `All MSBuild C++ projects...` message would be very good. In the case of partial VS2015 failure, perhaps something like `Visual Studio 2017 and later can now #include any installed libraries`.

Reply:
Done! Thanks for helping wording the messages.